state.go

```diff
@@ -397,6 +397,16 @@ func NewUpdateErrorState(err menderError, update UpdateResponse) State {
 func (ue *UpdateErrorState) Handle(c Controller) (State, bool) {
 	// TODO error handling
 	c.ReportUpdateStatus(ue.update, statusFailure)
+
+	// TODO fetch logs from logger
+	c.UploadLog(ue.update, []LogEntry{
+		LogEntry{
+			Timestamp: time.Now().Format(time.RFC3339),
+			Level:     "error",
+			Message:   "update failed",
+		},
+	})
```

Comment:
I've introduced `menderError` exactly for the purpose of distinguishing soft vs. hard errors (the error is considered transient unless `menderError.IsFatal()` returns true). This is also captured as part of error states data. IMO we can work on this, perhaps extend menderError with additional information.

Also, capturing the previous state information when transitioning to one of the error states might be useful in case we want to give special treatment to certain states that failed.